zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -117,6 +146,30 @@ impl Chain {
     pub fn is_empty(&self) -> bool {
         self.blocks.is_empty()
     }
+
+    pub fn mmr_hash(&self) -> ChainHistoryMmrRootHash {
+        self.mmr.hash()
+    }
+
+    /// Clone the Chain but not the MMR.
+    ///
+    /// Useful when forking, where the MMR is rebuilt anyway.
+    fn clone_with_mmr(&self, mmr: MerkleMountainRange) -> Self {
+        Chain {
+            blocks: self.blocks.clone(),
+            height_by_hash: self.height_by_hash.clone(),
+            tx_by_hash: self.tx_by_hash.clone(),
+            created_utxos: self.created_utxos.clone(),
+            spent_utxos: self.spent_utxos.clone(),
+            sprout_anchors: self.sprout_anchors.clone(),
+            sapling_anchors: self.sapling_anchors.clone(),
+            sprout_nullifiers: self.sprout_nullifiers.clone(),
+            sapling_nullifiers: self.sapling_nullifiers.clone(),
+            orchard_nullifiers: self.orchard_nullifiers.clone(),
+            partial_cumulative_work: self.partial_cumulative_work,
+            mmr,
+        }
+    }
```

Comment:
This is good, because we'll have to make an explicit decision for each new field.

I think `with_mmr` is the egonomic name here - because it's not exactly a clone, it's a clone-and-replace.

Here's a similar method that copies a date, but changes the timezone:
https://docs.rs/chrono/0.4.19/chrono/struct.Date.html#method.with_timezone

Reply:
Good idea!